### PR TITLE
Update pino to 2.12.3, also exposing internal interfaces.

### PR DESCRIPTION
**Pull request by Volox**

### CODE REVIEW

File: pino/pino-tests.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-/// <reference path="../node/node.d.ts"/>   // only for require()
+/// <reference path="../node/index.d.ts"/>   // only for require()
```
Comment (mhegazy) [inline pino/pino-tests.ts]: do not use `/// <reference path=".." />`, use `/// <reference types="node" />` instead

**Comment by mhegazy** [comment]
Also please address the merge conflicts

File: pino/pino.d.ts [hunk 2]
```diff
@@ -1,86 +1,90 @@
...
 // Definitions: https://github.com/borisyankov/DefinitelyTyped
 
-/// <reference path="../node/node.d.ts"/>   // only for require()
+/// <reference path="../node/index.d.ts"/>   // only for require()
 
 declare module 'pino' {
+    import stream = require('stream');
 
-    import stream = require('stream')
+    namespace pino {
```
Comment (Volox) [inline pino/pino.d.ts]: I followed the guidelines of typings for exposing a [function + utility](https://github.com/typings/typings/blob/master/docs/examples.md#function--utility).

**Comment by mhegazy** [comment]
one more round of merge conflicts.

**Comment by yuit** [comment]
@Volox  Thank you for the PR, could you fix the merge conflict?

**Comment by paulvanbrenk** [comment]
Closing this PR since it seem abandoned

File: pino/pino.d.ts (hunk 2 above)
Comment (psnider) [inline pino/pino.d.ts]: @Volox I just submitted an update to pino.d.ts to bring in line with the latest version.
See: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/13303

**Pull request closed without merging**